assets/monitoring/grafana/v1alpha1/registry.go

```diff
@@ -17,44 +18,68 @@ func ParseObjectTemplateOrDie[T runtime.Object](name, tmplString string) assets.
 	return assets.ParseObjectTemplateOrDie[T](name, tmplString, assets.TemplateFuncs, scheme.Codecs.UniversalDeserializer())
 }
 
+type Registry struct {
+	GrafanaDeploymentTemplate             assets.ObjectTemplate[*appsv1.Deployment]
```

Comment:
I wonder if we could just make tha lazy init per type with a generic wrapper and a lambda, so it all stays colocated - now the it's prone to forgetting to initialize the value.

Reply:
done. It required changing return value of ParseObjectTemplate to return pointer. Otherwise wrapped value couldn't be used directly due to RenderObject requiring pointer receiver.